### ISSUE TITLE
Allow syncing third-party plugins settings

### ISSUE THREAD

**Issue by crimoniv** [fman-community/fman#374]
This is somewhat related to #149 and #371,

When syncing settings across multiple machines -e.g., with GNU `stow` + `git`- there is no clean way to sync installed plugins and their settings. Right now I can only think of 2 ways:

- Sync also the folder `Plugins/Third-party` (It feels very dirty).
- Manually install and configure the plugins.

It would be great to be able to auto-install plugins from settings when **fman** starts, just like **Sublime Text** manages its plugins with the `Package Control.sublime-settings` file.

**Comment by vovcacik** [comment]
In the context of #371, I read it as if the fman's config dir will get split in two parts (*1.* local only part with all the stuff you don't wanna sync and *2.* synchronizable part) then you want:
- authoritative list of plugins in the synchronizable part
- automatically install and uninstall plugins in each fman instance based on the list

Right?

**Comment by mherrmann** [comment]
> In the context of #371, I read it as if the fman's config dir will get split in two parts (1. local only part with all the stuff you don't wanna sync and 2. synchronizable part)

I don't think I said that(?) Not to say that it wouldn't be a good solution; But I haven't thought about #371 enough yet to be able to guarantee this.

I understood it the same way, that @crimoniv wants a settings file that fman uses to automatically install plugins not present on the current machine.

**Comment by vovcacik** [comment]
> I don't think I said that(?) Not to say that it wouldn't be a good solution; But I haven't thought about #371 enough yet to be able to guarantee this.

That's how I interpreted `FMAN_USER_DIR` and `FMAN_DATA_DIR` in your comment [here](https://github.com/fman-users/fman/issues/371#issuecomment-383841981)

**Comment by mherrmann** [comment]
> That's how I interpreted `FMAN_USER_DIR` and `FMAN_DATA_DIR`

I see. I wrote there that `FMAN_USER_DIR` defaults to `%FMAN_DATA_DIR%\Plugins\User`. That directory already exists at the moment and in particular contains `Settings\Visited Paths.json`, which you most likely do not want to sync, even though it is in `FMAN_USER_DIR`. A new solution along the lines of what you have been thinking of @vovcacik would have to put `Visited Paths.json` somewhere outside `FMAN_USER_DIR`.

**Comment by crimoniv** [comment]
> I understood it the same way, that @crimoniv wants a settings file that fman uses to automatically install plugins not present on the current machine.

Yes, that is exactly what I am referring to. I only mean that current fman config already supports tools like `stow` to synchronize settings among multiple machines, it is just lacking a proper way to synchronize plugins (without needing to sync plugins' sources too).

But I understand that #371's shared config should also share the plugins' sources, so I've referenced it just as a reminder that we must reach a midway between these two options.

**Comment by mherrmann** [comment]
@crimoniv noted, thanks :-)

**Comment by mwean** [comment]
I'd love to see this too. The only way I can think to sync installed plugins without actually copying the files would be to store a list of git repos and install them in the correct location on all machines, but that's not a great solution. It would be nice if the list of plugins was stored in `User.json` or something and then fman auto-installs missing plugins.
